Add keepMounted option and fix screen glitch

pikaju opened the pull request:
Keeps the Modal mounted even when it is hidden. This is useful for keeping the DOM state inside the Modal as well as for SEO purposes.

Fixes #233 and #495

pradel commented on react-responsive-modal/styles.css:
```diff
@@ -92,8 +95,15 @@
+.react-responsive-modal-overlay,
+.react-responsive-modal-container,
+.react-responsive-modal-modal {
+  animation-fill-mode: forwards !important;
```
can you please explain why this is needed?

pikaju commented:
Hey @pradel, thanks for the response. I added an example and docs in the Next project.

I also changed how modals are centered, from the "inline" behavior to flex-box centering. I did this because there was an issue when the modal had little space (e.g. on a mobile device) and the pseudo-element you had before was wrapped to a new line. This implementation doesn't do that.

akoll commented on website/src/docs/index.mdx:
```diff
@@ -103,6 +104,26 @@ You can also set to trap focus within the modal, but decide where to put focus w
 
+### Keeping a hidden modal mounted
+
+By setting the `keepMounted` property to `true`, you can specify that the modal should always be mounted to the DOM, even when hidden it is hidden.
```
```suggestion
By setting the `keepMounted` property to `true`, you can specify that the modal should always be mounted to the DOM, even when it is hidden.
```

pradel commented:
@pikaju could you please split the pr into two parts?
- the keepMounted can be released as a minor version first
- then the change you made to the CSS for the mobile issue, this one will need to be released as a breaking change as it may conflict with the CSS used by users of this package

pikaju commented:
@pradel Sorry, but that doesn't seem worth the effort, I suggest just releasing a new major version
